onnxruntime/core/providers/dnnl/dnnl_common.h

```diff
@@ -65,9 +66,14 @@ class PrimitivePool {
  private:
   // For thread safety, the map needs to be kept in thread local storage.
   static inline std::unordered_map<std::string, std::unique_ptr<PrimitiveBase>>& GetMap() {
-    static thread_local std::unordered_map<std::string, std::unique_ptr<PrimitiveBase>> map;
-    return map;
+    static thread_local auto map = std::make_unique<std::unordered_map<std::string, std::unique_ptr<PrimitiveBase>>>();
+    static thread_local RunOnUnload run_on_unload{
+        [&map = map]() {
```

Comment:
I think this needs a comment explaining why it's needed and how it works.

Reply:
Ah, I put a comment in the declaration in the header. I can put a second one here saying to see the header?